lib/wallaby/browser.ex

```diff
@@ -527,13 +527,35 @@ defmodule Wallaby.Browser do
 
   @doc """
   Executes javascript synchoronously, taking as arguments the script to execute,
-  and optionally a list of arguments available in the script via `arguments`
+  an optional list of arguments available in the script via `arguments`, and an
+  optional callback function with the result of script execution as a parameter.
   """
+  @spec execute_script(parent, String.t) :: parent
   @spec execute_script(parent, String.t, list) :: parent
+  @spec execute_script(parent, String.t, ((binary()) -> any())) :: parent
+  @spec execute_script(parent, String.t, list, ((binary()) -> any())) :: parent
 
-  def execute_script(session, script, arguments \\ []) do
+
+  def execute_script(session, script) do
+    {:ok, _} = Driver.execute_script(session, script, [])
+    session
+  end
+
+  def execute_script(session, script, arguments) when is_list(arguments) do
+    {:ok, _} = Driver.execute_script(session, script, arguments)
+    session
+  end
+
+  def execute_script(session, script, callback) when is_function(callback) do
+    {:ok, value} = Driver.execute_script(session, script, [])
+    callback.(value)
+    session
+  end
```

Comment:
Fine by me, I'd just like to group the same functions (same arity) together (no newline inbetween)